README-ReactiveSwift.md

```diff
@@ -0,0 +1,355 @@
+
+ 1. [Introduction](#introduction)
+ 1. [Example: online search](#example-online-search)
+ 1. [How does ReactiveSwift relate to Rx?](#how-does-reactivecocoa-relate-to-rx)
+ 1. [Getting started](#getting-started)
+ 1.  [Playground](#playground)
+
+If you’re already familiar with functional reactive programming or what
+ReactiveSwift is about, check out the [Documentation][] folder for more in-depth
+information about how it all works. Then, dive straight into our [documentation
+comments][Code] for learning more about individual APIs.
+
+If you'd like to use ReactiveSwift with Apple's Cocoa frameworks,
+[ReactiveCocoa](https://github.com/ReactiveCocoa/ReactiveCocoa) provides
+extensions that work with ReactiveSwift.
+
+If you have a question, please see if any discussions in our [GitHub
+issues](https://github.com/ReactiveCocoa/ReactiveSwift/issues?q=is%3Aissue+label%3Aquestion+) or [Stack
+Overflow](http://stackoverflow.com/questions/tagged/reactive-cocoa) have already
+answered it. If not, please feel free to [file your
+own](https://github.com/ReactiveCocoa/ReactiveSwift/issues/new)!
+
+#### Compatibility
+
+This documents ReactiveSwift 3.x which targets `Swift 3.0.x`. For `Swift 2.x` support see [ReactiveCocoa
+4](https://github.com/ReactiveCocoa/ReactiveCocoa/tree/v4.0.0).
+
+## Introduction
+
+ReactiveSwift is inspired by [functional reactive
+programming](https://joshaber.github.io/2013/02/11/input-and-output/).
+Rather than using mutable variables which are replaced and modified in-place,
+RAC offers “event streams,” represented by the [`Signal`][Signals] and
+[`SignalProducer`][Signal producers] types, that send values over time.
+
+Event streams unify common patterns for asynchrony and event
+handling, including:
+
+ * Delegate methods
+ * Callback blocks
+ * Notifications
+ * Control actions and responder chain events
+ * [Futures and promises](https://en.wikipedia.org/wiki/Futures_and_promises)
+ * [Key-value observing](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/KeyValueObserving/KeyValueObserving.html) (KVO)
+
+Because all of these different mechanisms can be represented in the _same_ way,
+it’s easy to declaratively chain and combine them together, with less spaghetti
+code and state to bridge the gap.
+
+For more information about the concepts in ReactiveSwift, see the [Framework
+Overview][].
+
+## Example: online search
+
+Let’s say you have a text field, and whenever the user types something into it,
+you want to make a network request which searches for that query.
+
+#### Observing text edits
+
+The first step is to observe edits to the text field, using a RAC extension to
+`UITextField` specifically for this purpose:
+
+```swift
+let searchStrings = textField.rac_textSignal()
```

Comment:
I think we can clean this up later. This still demonstrates ReactiveSwift, even if it uses ReactiveCocoa. (And this is change is plenty big.)